src/test/java/com/meilisearch/integration/SearchNestedTest.java

```diff
@@ -0,0 +1,125 @@
+package com.meilisearch.integration;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+import com.meilisearch.integration.classes.AbstractIT;
+import com.meilisearch.integration.classes.TestData;
+import com.meilisearch.sdk.Index;
+import com.meilisearch.sdk.SearchRequest;
+import com.meilisearch.sdk.Settings;
+import com.meilisearch.sdk.Task;
+import com.meilisearch.sdk.json.GsonJsonHandler;
+import com.meilisearch.sdk.utils.Movie;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+
+@Tag("integration")
+public class SearchNestedTest extends AbstractIT {
+
+    private TestData<Movie> testData;
+
+    final class Results {
+        Movie[] hits;
+        int offset;
+        int limit;
+        int nbHits;
+        boolean exhaustiveNbHits;
+        int processingTimeMs;
+        String query;
+    }
+
+    @BeforeEach
+    public void initialize() {
+        this.setUp();
+        if (testData == null) testData = this.getTestData(NESTED_MOVIES, Movie.class);
+    }
+
+    @AfterAll
+    static void cleanMeiliSearch() {
+        cleanup();
```

Comment:
Hey, look here: https://docs.cypress.io/guides/references/best-practices#Using-after-or-afterEach-hooks

It is not so good to teardown your test setup after the execution :D

Reply:
I don't understand, do you mean I have to do it `afterEach` instead or not clean at all?
Because I like to clean after each test file in order to have independent test no?